FinSearchUnified/Controllers/Backend/FindologicStaging.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+use FinSearchUnified\Components\ConfigLoader;
+use Shopware\Components\CSRFWhitelistAware;
+use Shopware\Models\Config\Value;
+use Shopware\Models\Shop\Repository;
+use Shopware\Models\Shop\Shop;
+
+class Shopware_Controllers_Backend_FindologicStaging extends Shopware_Controllers_Backend_ExtJs implements
+    CSRFWhitelistAware
+{
+    public function getWhitelistedCSRFActions()
+    {
+        return [
+            'index'
+        ];
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function indexAction()
+    {
+        Shopware()->Plugins()->Controller()->ViewRenderer()->setNoRender(true);
+        $this->assertPluginIsActive();
+
+        $shopkey = Shopware()->Config()->offsetGet('ShopKey');
+        $this->assertShopKeyIsValid($shopkey);
+        $shop = $this->getCurrentShop($shopkey);
+        $isStaging = $this->isStaging();
+
+        $this->redirect($this->getRedirectUrl($isStaging, $shop));
+    }
+
+    private function assertPluginIsActive()
+    {
+        $isActive = Shopware()->Config()->offsetGet('ActivateFindologic');
+
+        if (!$isActive) {
+            throw new \RuntimeException('Please ensure the plugin is active!');
```

Comment:
Create own Exception classes for that and throw them here. If these exceptions are thrown, the message should be displayed in the frontend. e.g.

```php
throw new StagingModeException('Please ensure that the FinSearchUnified plugin is active.');
```

in the indexAction:

```php
try {

} catch (StagingModeException $e) {
    echo $e->getMessage();
}
```